### ISSUE TITLE
Queries of `rpm_packages` display no data

### ISSUE THREAD

**Issue by Neon-C23** [osquery/osquery#7925]
# Bug report
I deployed osqueryd on my entire machine and wanted it to be able to send some data to my kafka every day for auditing, here is my osquery.conf file:

{
  "options": {
    "logger_plugin": "kafka_producer",
    "logger_kafka_brokers": "soc-server-01:9092",
    "logger_kafka_topic": "agentquery",
    "logger_kafka_compression": "",
    "logger_kafka_acks": "all",
    "schedule_splay_percent": 1
  },
  "schedule": {
  },
  "packs": {
    "agent": {
      "queries": {
    "system_info_n": {
      "query": "select 'system_info' as MARK, '192.168.1.21' as HOST,* from system_info;",
      "interval": 86400,
      "snapshot": true
    },
    "interface_addresses_n": {
      "query": "select 'interface_addresses' as MARK, '192.168.1.21' as HOST,* from interface_addresses;",
      "interval": 86400,
      "snapshot": true
    },
    "os_version_n": {
      "query": "select 'os_version' as MARK, '192.168.1.21' as HOST,* from os_version;",
      "interval": 86400,
      "snapshot": true
    },
    "users_n": {
      "query": "select 'users' as MARK, '192.168.1.21' as HOST,* from users;",
      "interval": 86400,
      "snapshot": true
    },
    "groups_n": {
      "query": "select 'groups' as MARK, '192.168.1.21' as HOST,g.*,u.username from groups g left join users u on g.g
id=u.gid;",
      "interval": 86400,
      "snapshot": true
    },
    "rpm_packages_n": {
      "query": "select 'rpm_packages' as MARK, '192.168.1.21' as HOST,name,version,source,size from rpm_packages;",
      "interval": 86400,
      "snapshot": true
    },
    "processes_n": {
      "query": "select 'processes' as MARK, '192.168.1.21' as HOST,p.pid,p.name,p.state,p.uid,p.on_disk,p.start_time,
p.disk_bytes_read,p.disk_bytes_written,p.parent,p.user_time,p.system_time,p.resident_size,p.total_size,p.path,p.cmdli
ne,u.username from processes p left join users u on p.uid=u.uid where p.parent!=2;",
      "interval": 300,
      "snapshot": true
    },
    "listening_ports_n": {
      "query": "select 'listening_ports' as MARK, '192.168.1.21' as HOST,l.pid,l.address,l.port,l.protocol,p.name,p.p
ath,p.cmdline from listening_ports l left join processes p on l.pid=p.pid where l.port>0 and l.address!='127.0.0.1';"
,
      "interval": 300,
      "snapshot": true
    },
    "process_open_sockets_n": {
      "query": "select 'process_open_sockets' as MARK, '192.168.1.21' as HOST,s.pid,p.name,p.path,protocol,local_addr
ess,remote_address,local_port,remote_port,s.state from process_open_sockets s left join processes p on s.pid=p.pid wh
ere local_port>0 and remote_port >0 and local_address!=remote_address and s.state!='TIME_WAIT' and s.pid>0;",
      "interval": 300,
      "snapshot": true
    },
    "crontab_n": {
      "query": "select 'crontab' as MARK, '192.168.1.21' as HOST,* from crontab;",
      "interval": 86400,
      "snapshot": true
    }
      }
    }
  },
  "kafka_topics": {
    "agentquery":  ["pcak_agent_system_info_n"],
    "agentquery":  ["pack_agent_interface_addresses_n"],
    "agentquery":  ["pack_agent_os_version_n"],
    "agentquery":  ["pack_agent_users_n"],
    "agentquery":  ["pack_agent_groups_n"],
    "agentquery":  ["pack_agent_rpm_packages_n"],
    "agentquery":  ["pack_agent_processes_n"],
    "agentquery":  ["pack_agent_listening_ports_n"],
    "agentquery":  ["pack_agent_process_open_sockets_n"],
    "agentquery":  ["pack_agent_crontab_n"]
  }
}

At first, osqueryd worked normally, but after running for half a month, it was found that kafka no longer received data from rpm_packages tables, but other tables still worked normally. 

I initially suspected that the problem was caused by a corrupted rpmdb, but after checking it found that it didn't. 

Then I deleted the osquery .db folder and tried restarting osqueryd, which didn't solve the problem either. 

And then I found that using 'osqueryi' or 'osqueryd -s' to query rpm_packages table still didn't have any data.

### What operating system and version are you using?
CentOS Linux 7.6

### What version of osquery are you using?
5.0.1

### What steps did you take to reproduce the issue?
It's not clear to me if I was able to reproduce this issue.
### What did you expect to see?
osqueryd and osqueryi can query data from rpm_packages tables.
### What did you see instead?
The query structure shows no data.

**Comment by NeonC135** [comment]
This is me performing "osqueryd -s" select * from rpm_packages;" --verbose" result:

I0131 12:10:14.159471 516105 init.cpp:357] osqueryd initialized [version=5.0.1-7-gb808557-dirty]
I0131 12:10:14.159521 516105 extensions.cpp:453] Could not autoload extensions: Cannot open file for reading: /etc/osquery/extensions.load
I0131 12:10:14.159567 516105 init.cpp:568] An error occurred during extension manager startup: Extensions disabled
I0131 12:10:14.159586 516105 auto_constructed_tables.cpp:97] Removing stale ATC entries
I0131 12:10:14.160133 516105 events.cpp:70] Skipping subscriber: process_file_events: Subscriber disabled via configuration
I0131 12:10:14.181478 516105 glog_logger.cpp:49] RPM notice: Failed to read auxiliary vector, /proc not mounted?
I0131 12:10:14.182937 516105 glog_logger.cpp:49] RPM notice: cannot open Packages database in 
I0131 12:10:14.183066 516105 dispatcher.cpp:149] Thread: 140076145919680 requesting a stop
I0131 12:10:14.183080 516105 dispatcher.cpp:122] Thread: 140076145919680 requesting a join
I0131 12:10:14.183092 516105 dispatcher.cpp:144] Services and threads have been cleared

**Comment by Neon-C23** [comment]
I restarted a server today that wasn't able to query the rpm_packages table, after which I was surprised to find that kafka was actually picking up data from the rpm_packages table on the restarted machine.

I then tried to reboot another machine with the same problem, and happily the reboot system worked, and the rpm_packages table on this machine is back to normal.

Does this mean that this is not a program error? Rather, it's a system or rpm problem?

Anyway, I think I should close this issue now.

**Comment by Neon-C23** [comment]
I restarted a server today that wasn't able to query the rpm_packages table, after which I was surprised to find that kafka was actually picking up data from the rpm_packages table on the restarted machine.

I then tried to reboot another machine with the same problem, and happily the reboot system worked, and the rpm_packages table on this machine is back to normal.

Does this mean that this is not a program error? Rather, it's a system or rpm problem?

Anyway, I think I should close this issue now.
